eq3bt/eq3btsmart.py

```diff
@@ -201,8 +211,12 @@ def update(self):
         value = struct.pack('BBBBBBB', PROP_INFO_QUERY,
                             time.year % 100, time.month, time.day,
                             time.hour, time.minute, time.second)
-
-        self._conn.make_request(PROP_WRITE_HANDLE, value)
+        try:
+            self._conn.make_request(PROP_WRITE_HANDLE, value)
+        except Exception as ex:
```

Comment:
I don't currently have a test device at hand to check this, but wouldn't it make more sense to perform the error&retry checking inside `make_request` or leave the whole re-try machinery for upstreams to handle?

Reply:
Inside the make_request function (connection.py) possible errors are fetched. But the calling instance does not recognize and handle the error. That's the reason for the stacktrace.

You don't need a real test device, try it with a random mac like 11:22:33:44:55:66.

Reply:
Okay, I found a test device and did some quick testing.

Now, I think it's better to let the exceptions bubble up instead of having a device global error flag (e.g., if you query multiple different things, some of them may succeed and you can still display that information to the user), and handle the exception at user's end.

In the cli, instead of checking for the the error flag, it would be better to do something like this for the `@cli.group` to capture all exceptions: https://stackoverflow.com/questions/44344940/python-click-subcommand-unified-error-handling/44347763#44347763